crates/syntax/rust.ungram

```diff
@@ -290,7 +290,7 @@ TypeParam =
 
 ConstParam =
   Attr* 'const' Name ':' Type
-  ('=' default_val:Expr)?
+  ('=' default_val:ConstArg)?
```

Comment:
while [the parser is aware](https://github.com/rust-lang/rust-analyzer/blob/master/crates/parser/src/grammar/generic_args.rs#L129-L168) of the restrictions for `ConstArg`, rust.ungram [is not](https://github.com/rust-lang/rust-analyzer/blob/master/crates/syntax/rust.ungram#L61-L62). not sure if it's worth fixing